src/js/mediaStore.js

```diff
@@ -0,0 +1,1241 @@
+import { document, globalThis } from './utils/server-safe-globals.js';
+import {
+  AvailabilityStates,
+  MediaUIEvents,
+  StreamTypes,
+  TextTrackKinds,
+  TextTrackModes,
+} from './constants.js';
+import { getTextTracksList, updateTracksModeTo } from './utils/captions.js';
+import { containsComposedNode } from './utils/element-utils.js';
+import { fullscreenApi } from './utils/fullscreen-api.js';
+import {
+  airplaySupported,
+  castSupported,
+  fullscreenSupported,
+  hasFullscreenSupport,
+  hasPipSupport,
+  hasVolumeSupportAsync,
+  pipSupported,
+} from './utils/platform-tests.js';
+
+/**
+ * @typedef {'on-demand'|'live'|'unknown'} StreamTypeValue
+ */
+
+/**
+ * @typedef {'unavailable'|'unsupported'} AvailabilityTypeValue
+ */
+
+/**
+ * @typedef {Partial<HTMLVideoElement> & Pick<HTMLMediaElement,'play'> & {
```

Comment:
Does the usage of `Pick` here make `play` a required property of the combined type? wondering if we need something like `Partial<Pick<HTMLMediaElement,'play'>>` to make it optional.

Reply:
Within the team, including discussions with @heff, we've had some soft rules around minimal requirements for what we've historically called "a slotted media element" in Media Chrome (and what is here called a "media state owner"). The tacit place we've landed for now is something like:
1. for something to count as a media state owner, it must minimally have
    1. a `play()` method that must conform to the `HTMLMediaElement` public API
    2. a `paused` property that must conform to the `HTMLMediaElement` public API (I should add this to the type definition, so I'm glad you called this out.
    3. 1+ event that's used by our generic/default state facade to monitor for `paused` state changes (I should add the add/remove event handlers as required as well here). The most obvious candidates would be `play` and `pause` (See descriptions in https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement#events as well as the algorithms/steps defined in https://html.spec.whatwg.org/multipage/media.html#playing-the-media-resource
2. _**if**_ a media state owner has a given property or method that exists on the extended set of keys defined in this type def, it must conform to the `HTMLMediaElement` public API or the corresponding extended properties as defined in the `typedef` TypeScript `type`.